OpenRA.Game/ObjectCreator.cs

```diff
@@ -54,7 +55,13 @@ public ObjectCreator(Manifest manifest, FileSystem.FileSystem modFiles)
 				Assembly assembly;
 				if (!ResolvedAssemblies.TryGetValue(hash, out assembly))
 				{
-					assembly = Assembly.Load(data);
+					Stream debugStream = null;
+					if (Type.GetType("Mono.Runtime") != null && modFiles.TryOpen(path + ".mdb", out debugStream))
+						assembly = Assembly.Load(data, debugStream.ReadAllBytes());
+					else if (modFiles.TryOpen(path.Substring(0, path.Length - 4) + ".pdb", out debugStream))
```

Comment:
My preference would be to define `isMonoRuntime = Type.GetType("Mono.Runtime") != null` in the constructor, and here do something like:

```csharp
Stream symbolStream = null;
var hasSymbols = false;
if (usingMono)
	hasSymbols = modFiles.TryOpen(path + ".mdb", out symbolStream));

// Never mono versions can also load newer format .pdb files
if (!hasSymbols)
	hasSymbols = modFiles.TryOpen(path.Substring(0, path.Length - 4) + ".pdb", out symbolStream);

assembly = hasSymbols ? Assembly.Load(data, symbolStream.ReadAllBytes()) : Assembly.Load(data);


```